include/rcpputils/filesystem_helper.hpp

```diff
@@ -536,6 +537,62 @@ inline bool remove(const path & p)
 #endif
 }
 
+/**
+ * \brief Remove the directory at the path p and its content.
+ *
+ * Additionally to \sa remove, remove_all removes a directory and its containing files.
+ *
+ * \param The path of the directory to remove.
+ * \return true if the directory exists and it was successfully removed, false otherwise.
+ */
+inline bool remove_all(const path & p)
+{
+  if (!is_directory(p)) {return false;}
+
+#ifdef _WIN32
+  // We need a string of type PCZZTSTR, which is a double null terminated char ptr
+  size_t length = p.string().size();
+  TCHAR * temp_dir = new TCHAR[length + 2];
+  memcpy(temp_dir, p.string().c_str(), length);
+  temp_dir[length] = 0;
+  temp_dir[length + 1] = 0;    // double null terminated
+
+  SHFILEOPSTRUCT file_options;
+  file_options.hwnd = nullptr;
+  file_options.wFunc = FO_DELETE;    // delete (recursively)
+  file_options.pFrom = temp_dir;
+  file_options.pTo = nullptr;
+  file_options.fFlags = FOF_NOCONFIRMATION | FOF_SILENT;    // do not prompt user
+  file_options.fAnyOperationsAborted = FALSE;
+  file_options.lpszProgressTitle = nullptr;
+  file_options.hNameMappings = nullptr;
+
+  SHFileOperation(&file_options);
```

Comment:
check return value? check fAnyOperationsAborted status?
See https://docs.microsoft.com/en-us/windows/win32/api/shellapi/nf-shellapi-shfileoperationa#return-value (I guess SHFileOperation is the same that SHFileOperationA :joy:)